rust/noosphere-ns/src/records.rs

```diff
@@ -0,0 +1,320 @@
+use anyhow::{anyhow, Error};
+use cid::Cid;
+use serde::{
+    de::{self, Deserializer, MapAccess, SeqAccess, Visitor},
+    ser::{SerializeStruct, Serializer},
+    Deserialize, Serialize,
+};
+
+use std::{
+    fmt,
+    str::FromStr,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+/// An [NSRecord] is a struct representing a record stored in the
+/// Noosphere Name System's DHT containing a [Cid] of the
+/// result, as well as a TTL expiration as seconds from Unix epoch.
+///
+/// # Serialization
+///
+/// When transmitting records across the network, they're encoded
+/// as JSON-formatted UTF-8 strings. The [NSRecord::to_bytes]
```

Comment:
From other comment: No opinion I think on the JSON representation, just the quick and easy solution for now. If we do not need to "staple" NS data onto the record, this could simply be utf8 encoded Ucan

Additionally: the data could be CBOR-ified as well rather than a base64 encoding